src/codecs/encoding/transformer.rs

```diff
@@ -0,0 +1,360 @@
+#![deny(missing_docs)]
```

Comment:
In the future, it would be useful to call out which code was just moved to try to focus reviews. This could be done in the form of PR comments or separating out refactoring commits.

Reply:
True - I wasn't very considerate about that! Would save quite some time.

Reply:
No worries! I checked first to see if this was new code and found it had just been moved.